pkgs/development/python-modules/detectron2/default.nix

```diff
@@ -79,14 +83,27 @@ buildPythonPackage {
     iopath
     omegaconf
     hydra-core
-    black
     packaging
-  ] ++ optional-dependencies.all;
+    torch
+    pydot
+  ];
 
   passthru.optional-dependencies = optional-dependencies;
 
-  # disable the tests for now until someone can check on a linux machine.
-  # doCheck = false;
+  nativeCheckInputs = [
+    pytestCheckHook
+    black
+    torchvision
+  ];
+
+  preCheck = ''
+    substituteInPlace setup.py --replace "black==" "black>="
+    cd tests
+  '';
+
```

Comment:
```suggestion
```
I tried to add some changes to make the test work, but I could not run it because it would try to download data or hang.
I don't think these are necessary, as they are incomplete anyway.

Reply:
Would you mind leaving this though just in case someone ever comes around to actually fix the tests ?

Reply:
That someone is here.